src/index.js

```diff
@@ -0,0 +1,73 @@
+const gridRows = 20;
+const gridCols = 10;
+const gameGrid = Array.from({ length: gridRows }, () => Array(gridCols).fill(0));
+
+const tetrisPiece = [ // creates all possible pieces and various colours
+    { shape: [[1, 1, 1, 1]], colour: 'blue' },
+    { shape: [[1, 1], [1, 1]], colour: 'yellow' },
+    { shape: [[1, 1, 1], [1, 0, 0]], colour: 'pink' },
+    { shape: [[1, 1, 1], [0, 0, 1]], colour: 'red' },
+    { shape: [[1, 1, 1], [0, 1, 0]], colour: 'green' },   
+    { shape: [[1, 1, 0], [0, 1, 1]], colour: 'maroon' },
+    { shape: [[0, 1, 1], [1, 1, 0]], colour: 'orange' }
+];
+
+function getrandompiece() { // creates a random piece and returns it
+    const randomInt = Math.floor(Math.random() * tetrisPiece.length);
+    return tetrisPiece[randomInt];
+}
+
+const randomPiece = getrandompiece(); // randomizes each piece and corresponding
+console.log("random tetris piece:", randomPiece.shape);
+console.log("random tetris colour:", randomPiece.colour);
+let randomPieceColour = randomPiece.colour;
+
+function placePiece(cols, rows, piece, gameGrid) { // places a piece in the game grid
+    for (let i = 0; i < piece.shape.length; i++) {
+        for (let j = 0; j < piece.shape[i].length; j++) {
+            if (piece.shape[i][j]) {
+                gameGrid[rows + i][cols + j] = colourLetter(randomPiece); // Correctly place the piece so that where block occupies is first letter of colour
+            }
+        }
+    }
+}
+
+function colourLetter(randomPiece) { 
+    let firstLetter = randomPieceColour[0];
+    return firstLetter
+}
```

Comment:
This function uses a reference to `randomPieceColour`. This can work, though in JavaScript it's normally discouraged practice to do this, as it's not immediately clear where the reference to `randomPieceColour` is (in a big project it could be many 100s of lines away!)

Normally it's better to write functions in a pure way when possible, which looks like what this function is designed to do (input a piece, return a colour letter, without any side effects). Looks like it should be able to replace that reference to `randomPieceColour` with one to `randomPiece`